elements/lisk-dpos/test/unit/vote_weight_snapshot.spec.ts

```diff
@@ -0,0 +1,870 @@
+				);
+				expect(
+					additionalDelegates.every((delegate: Account) =>
+						voteWeights[0].delegates.find(
+							(d: Account) => d.address === delegate.address,
+						),
+					),
+				).toBeTrue();
+			});
+		});
+
+		describe('when there are non-self voting delegate within the top active delegates', () => {
+			let additionalDelegates: Account[];
+			let nonSelfVotedDelegate: Account;
+
+			beforeEach(async () => {
+				delegates = getDelegateAccounts(101);
+				for (const delegate of delegates) {
+					delegate.totalVotesReceived = randomBigInt(3000, 5000);
+					delegate.votes.push({
+						delegateAddress: delegate.address,
+						amount: delegate.totalVotesReceived,
+					});
+				}
+				nonSelfVotedDelegate = delegates[0];
+				// Update not to self vote
+				(nonSelfVotedDelegate.votes[0] as any).delegateAddress = '123L';
+				additionalDelegates = getDelegateAccounts(300);
+				for (const delegate of additionalDelegates) {
+					delegate.totalVotesReceived = randomBigInt(1000, 2999);
+					delegate.votes.push({
+						delegateAddress: delegate.address,
+						amount: delegate.totalVotesReceived,
+					});
+				}
+				block = {
+					id: 'random-block',
+					timestamp: 10100,
+					height: 1030,
+					generatorPublicKey: forgers[0].publicKey,
+					reward: BigInt(500000000),
+					totalFee: BigInt(100000000),
+					transactions: [],
+				} as Block;
+				chainStub.dataAccess.getDelegates.mockResolvedValue([
+					...delegates,
+					...additionalDelegates,
+				]);
+
+				// Setup for missed block calculation
+				const forgedBlocks = forgers
+					.map((forger, i) => ({
+						generatorPublicKey: forger.publicKey,
+						height: 928 + i,
+					}))
+					.slice(0, 102);
+				chainStub.dataAccess.getBlockHeadersByHeightBetween.mockResolvedValue(
+					forgedBlocks,
+				);
+
+				// TODO: Remove this forgers list mock after new DPoS implementation
+				const mockedForgersList = JSON.stringify([
+					{
+						round: 10,
+						delegates: [...forgers.map(d => d.publicKey).slice(0, 102)],
+					},
+				]);
+
+				stateStore = new StateStoreMock([forgers[0]], {
+					[CONSENSUS_STATE_FORGERS_LIST_KEY]: mockedForgersList,
+				});
+			});
+
+			it('should not include the non self voted delegate', async () => {
+				// Act
+				await dpos.apply(block, stateStore);
+
+				// Assert
+				const voteWeightsStr = await stateStore.consensus.get(
+					CONSENSUS_STATE_VOTE_WEIGHTS_KEY,
+				);
+				const voteWeights = JSON.parse(voteWeightsStr as string);
+				expect(voteWeights).toHaveLength(1);
+				expect(voteWeights[0].round).toEqual(13);
+				expect(
+					voteWeights[0].delegates.find(
+						(d: Account) => d.address === nonSelfVotedDelegate.address,
+					),
+				).toBeUndefined();
+				expect(voteWeights[0].delegates).toHaveLength(
+					delegates.length + additionalDelegates.length - 1,
+				);
+			});
+		});
+
+		describe('when there are banned delegates within the top active delegates', () => {
```

Comment:
```suggestion
		describe('when there are banned delegates', () => {
```

Reply:
as above, I find the formulation "banned within the top active" a bit weird. You cannot be banned and be part ot the active... but if you're banned, then you shouldn't be selected (for this the test is fine)

Reply:
same as above